### ISSUE TITLE
Add confirmation dialog for remote's power-off button.

### ISSUE THREAD

**Issue by GoogleCodeExporter** [peteaky/android-xbmcremote#228]
```
Feature :

The power button in the remote, landscape view, is easy to accidentally tap
so XBMC shuts down. So I wonder if anyone wants to make a delay for that
button. A 1 sec delay ore something so accidentally taping being ignored.
```

Original issue reported on code.google.com by `bolin.andreas@gmail.com` on 21 Apr 2010 at 9:18

**Comment by GoogleCodeExporter** [comment]
```
Usability-wise, long-press is a no-go. We can disuss a dialog.
```

Original comment by `phree...@gmail.com` on 21 Apr 2010 at 9:29
- Changed title: **Add confirmation dialog for remote's power-off button.**
- Changed state: **Accepted**
- Added labels: **Milestone-Release1.1**

**Comment by GoogleCodeExporter** [comment]
Original comment by `phree...@gmail.com` on 22 Apr 2010 at 10:03
- Added labels: **Type-Enhancement**
- Removed labels: **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
```
I hate dialogs. My TV remote works this way, even my grandma understand it.
```

Original comment by `bolin.andreas@gmail.com` on 29 Apr 2010 at 7:06

**Comment by GoogleCodeExporter** [comment]
```
how often do tv remotes put the power button way down in the middle of 
everything where you're likely to bump it?

i don't know about your xbmc boxes, but my android xbmc remote's power button 
won't turn it on. That means every time I accidentally hit the power button I 
have to get up, cross the room, and turn the PC back on.
```

Original comment by `ktwomb...@gmail.com` on 5 Nov 2010 at 11:28

**Comment by GoogleCodeExporter** [comment]
```
Also, the power-off button in landscape view is where the back button in 
portrait view is. And that is one of the most used buttons for me. In the last 
year I think I've wanted to power off the machine approximately once, but I'm 
accidentally hitting the landscape power-off button almost once per week.

In a similar vein, the "Exit XBMC" item from the menu does not actually exit 
XBMC, but also powers off the machine. And if you want a rationale on why 
adding another dialog isn't so bad: "Update library" from the menu in TV 
Shows/Movies pops up a scary confirmation dialog, even though a simple rescan 
isn't even a tenth as disruptive as powering off the machine.

And if you want to avoid adding more code complexity to the remote: pressing 
's' already pops up a dialog on the XBMC, with no additional work on the remote 
side.

So, in conclusion, suggestion for this behaviour: Make all the power off 
buttons (both in portrait and in landscape mode) act as "Press 's'". Rename 
"Exit XBMC" in the menu to "Power off XBMC" and optionally add a confirmation 
dialog.

(I think completely dropping a power off option that operates without an 
on-XBMC prompt would be too much, since I imagine there are people operating 
their XBMC headless/with disabled screen who would want such an option. But 
they most certainly are not the majority, and it doesn't need to be on the main 
remote control screen.)
```

Original comment by `hen...@ploetzli.ch` on 27 Mar 2011 at 5:25

**Comment by GoogleCodeExporter** [comment]
```
I agree that power-off should be out of the normal reach of our fingers and 
thumbs. Just like most remotes (upper corners).
It seems easy enough to include the text "hold to" the "power off" button would 
do it without need for any dialog--as well as avoiding many accidental short 
taps there.

```

Original comment by `munha...@gmail.com` on 7 Oct 2011 at 12:31
